src/solvers/petsc_dm_wrapper.C

```diff
@@ -99,6 +99,71 @@ void PetscDMWrapper::build_section( const System & system, PetscSection & sectio
   STOP_LOG ("build_section()", "PetscDMWrapper");
 }
 
+void PetscDMWrapper::build_sf( const System & system, PetscSF & star_forest )
+{
+  START_LOG ("build_sf()", "PetscDMWrapper");
+
+  const DofMap & dof_map = system.get_dof_map();
+
+  const std::vector<dof_id_type> & send_list = dof_map.get_send_list();
+
+  // Number of ghost dofs that send information to this processor
+  PetscInt n_leaves = send_list.size();
+
+  // Number of local dofs, including ghosts dofs
+  PetscInt n_roots = dof_map.n_local_dofs();
+  n_roots += send_list.size();
+
+  // This is the vector of dof indices coming from other processors
+  // TODO: We do a stupid copy here since we can't convert an
+  //       unsigned int* to PetscInt*
```

Comment:
Don't we static_assert that dof_id_type and PetscInt are the same size?  You ought to be able to reinterpret_cast send_list.data(), right?  Maybe add another static_assert here if you're feeling paranoid.

We do a copy in PetscVector::localize, but that's just because we're combining the send list with the local indices anyway.

Reply:
Damn. I really need to get more familiar with C++11.